test/conftest.py

```diff
@@ -0,0 +1,49 @@
+import os
+import subprocess
+from typing import Iterator
+
+import pytest
+
+from cockpit._vendor import systemd_ctypes
+
+
+# run tests on a private user bus
+@pytest.fixture(scope='session', autouse=True)
+def mock_session_bus(tmp_path_factory: pytest.TempPathFactory) -> Iterator[None]:
+    # make sure nobody opened the user bus yet...
+    assert systemd_ctypes.Bus._default_user_instance is None
+
+    tmpdir = tmp_path_factory.getbasetemp()
+    dbus_config = tmpdir / 'dbus-config'
+    dbus_addr = f'unix:path={tmpdir / "dbus_socket"}'
+
+    dbus_config.write_text(fr"""
+      <busconfig>
+        <fork/>
+        <type>session</type>
+        <listen>{dbus_addr}</listen>
+        <policy context="default">
+          <!-- Allow everything to be sent -->
+          <allow send_destination="*" eavesdrop="true"/>
+          <!-- Allow everything to be received -->
+          <allow eavesdrop="true"/>
+          <!-- Allow anyone to own anything -->
+          <allow own="*"/>
+        </policy>
+      </busconfig>
+    """)
+    try:
+        dbus_daemon = subprocess.run(
+            ['dbus-daemon', f'--config-file={dbus_config}', '--print-pid'], stdout=subprocess.PIPE
+        )
+    except FileNotFoundError:
+        yield None  # no dbus-daemon?  Don't patch.
+        return
+
+    pid = int(dbus_daemon.stdout)
+    os.environ['DBUS_SESSION_BUS_ADDRESS'] = dbus_addr
```

Comment:
We actually do test the system bus.

The story of this whole thing goes something like so:
 - for the test-server cases we export these mock objects on the user's session bus
 - but if we're running more than one copy of the tests then we're gonna get in trouble because we try to own the same name twice
 - that includes the case where we run the tests in parallel under xdist (ie: `pytest -n`)
 - so, solution: per-session mock session dbus

There's a wrinkle, though: we need to make sure that we set this environment variable before the first time that systemd_ctypes get used in the process, otherwise it will set up its "default user dbus" shared instance thing before this variable is set.  That's why this lives in the toplevel, although it's not like anything in `verify/` uses this either...

Some of the other tests already talk to the system D-Bus but don't do anything like owning names on it.  I think I did that back in the day because the system bus was more reliably available in our various test environments.